### PR TITLE
[pt][quant] Support either min or max in qclamp

### PR DESCRIPTION
**Pull request by dskhudia**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45937 [pt][quant] Support either min or max in qclamp**

torch.clamp can now be used with quantized tensors with either min argument or max argument only

Fixes https://github.com/pytorch/pytorch/issues/45928

Differential Revision: [D24153431](https://our.internmc.facebook.com/intern/diff/D24153431/)

### CODE REVIEW

File: aten/src/ATen/native/quantized/cpu/qclamp.cpp [hunk 1]
```diff
@@ -84,14 +86,18 @@ Tensor quantized_clamp_impl(
   Tensor qy;
   if (min && max) {
 #ifdef USE_PYTORCH_QNNPACK
-    if (at::globalContext().qEngine() == at::QEngine::QNNPACK && qx.scalar_type() == kQUInt8) {
+    if (at::globalContext().qEngine() == at::QEngine::QNNPACK &&
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cpu/qclamp.cpp:89]: qnnpack does not support the other two cases, can you keep the original error for qnnpack?

**Review by jerryzh168 — APPROVED** [review]
Thanks! please keep the original error for qnnpack.

File: aten/src/ATen/native/quantized/cpu/qclamp.cpp (hunk 1 above)
Comment (dskhudia) [reply aten/src/ATen/native/quantized/cpu/qclamp.cpp:89]: Does the following look ok?

Comment (jerryzh168) [reply aten/src/ATen/native/quantized/cpu/qclamp.cpp:89]: maybe you can separate the checks for qnnpack and fbgemm to make it more clear..

Comment (jerryzh168) [reply aten/src/ATen/native/quantized/cpu/qclamp.cpp:89]: i.e. don't repeat USE_PYTORCH_QNNPACK

File: aten/src/ATen/native/quantized/cpu/qclamp.cpp [hunk 2]
```diff
@@ -84,14 +86,23 @@ Tensor quantized_clamp_impl(
...
+        qx.scalar_type() == kQUInt8) {
       return qnnpack_clamp(qx, *min, *max);
     }
 #endif
     qclamp_stub(qx.device().type(), qx, *min, *max, qy);
   } else {
-    TORCH_CHECK(
-        false, "Both min and max should be specified for quantized clamp!");
+    if (at::globalContext().qEngine() == at::QEngine::QNNPACK) {
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/cpu/qclamp.cpp:96]: maybe add a USE_PYTORCH_QNNPACK macro here
```
#ifdef USE_PYTORCH_QNNPACK
    if (at::globalContext().qEngine() == at::QEngine::QNNPACK) {
     ...
   }
#endif
// fallback to fbgemm
if (max) {
} else if (min) {
} else {
}
```

**Pull request closed without merging**